lib/nidobata.rb

```diff
@@ -15,11 +15,20 @@ def init
       email = ask('Email:')
       password = ask('Password:', echo: false)
       data = {grant_type: 'password', username: email, password: password}
-
-      token = JSON.parse(http.post('/oauth/token', data.to_json, {'Content-Type' => 'application/json'}).tap(&:value).body)['access_token']
-      netrc = Netrc.read
-      netrc[IDOBATA_URL.host] = email, token
-      netrc.save
+      res = http.post('/oauth/token', data.to_json, {'Content-Type' => 'application/json'})
+
+      case res
+      when Net::HTTPSuccess
+        token = JSON.parse(res.tap(&:value).body)['access_token']
+        netrc = Netrc.read
+        netrc[IDOBATA_URL.host] = email, token
+        netrc.save
+      when Net::HTTPUnauthorized
```

Comment:
You can rescue this error on rescue clauses for this method as `Net::HTTPResponse#value` raises error when response is not a `Net::HTTPSuccess`.